README.md

```diff
@@ -210,11 +210,17 @@ export default Ember.Controller.extend({
   session: Ember.inject.service('session'),
 
   actions: {
-    authenticate() {
+    async authenticate() {
       let { identification, password } = this.getProperties('identification', 'password');
-      this.get('session').authenticate('authenticator:oauth2', identification, password).catch((reason) => {
+      try {
+        await this.session.authenticate('authenticator:oauth2', identification, password);
+      } catch(e) {
         this.set('errorMessage', reason.error || reason);
-      });
+      }
+
+      if (this.session.isAuthenticated) {
+        // What to do with all this success?
+      }
```

Comment:
This *could* be in the `try` as in the case that authentication fails, an error would be thrown (which is not a great API 😞).